patroni/postgresql/postmaster.py

```diff
@@ -102,15 +108,23 @@ def signal_stop(self, mode):
         if self.is_single_user:
             logger.warning("Cannot stop server; single-user server is running (PID: {0})".format(self.pid))
             return False
-        try:
-            self.send_signal(STOP_SIGNALS[mode])
-        except psutil.NoSuchProcess:
-            return True
-        except psutil.AccessDenied as e:
-            logger.warning("Could not send stop signal to PostgreSQL (error: {0})".format(e))
-            return False
-
-        return None
+        if os.name != "nt":
+            try:
+                self.send_signal(STOP_SIGNALS[mode])
+                return None
+            except psutil.NoSuchProcess:
+                return True
+            except psutil.AccessDenied as e:
+                logger.warning("Could not send stop signal to PostgreSQL (error: {0})".format(e))
+                return False
+        else:
+            if subprocess.call(['pg_ctl', 'kill', STOP_SIGNALS[mode], self.pid]) == 0:
```

Comment:
Usually `pg_ctl` is not in the default path, you have to use [pg_ctl method](https://github.com/zalando/patroni/blob/master/patroni/postgresql/__init__.py#L160-L166) instead.
Another problem is with passing `self.pid`, which is an integer, while command-line arguments could only be strings.

Reply:
But `pg_ctl kill` cannot have `['-D', self._data_dir]`, but they are hardcoded in pg_ctl method. Rewrite it?

Reply:
Oh, You are right.

Than `subprocess.call(['pg_ctl', 'kill', STOP_SIGNALS[mode], self.pid])` should be changed to:
```
subprocess.call([pgcommand('pg_ctl'), 'kill', STOP_SIGNALS[mode], str(self.pid)])
```

Or maybe just pass `pg_ctl` as an argument to the `signal_stop()`. It is kind of ugly, but we already doing it with `PostmasterProcess.start()`